src/util/psbt/serialize.rs

```diff
@@ -89,6 +89,30 @@ impl Deserialize for PublicKey {
     }
 }
 
+impl Serialize for EcdsaSig {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf = Vec::with_capacity(72);
+        buf.extend(self.sig.serialize_der().iter());
+        buf.push(self.hash_ty as u8);
+        buf
+    }
+}
+
+impl Deserialize for EcdsaSig {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        let (sighash_byte, signature) = bytes.split_last()
+            .ok_or(encode::Error::ParseFailed("empty partial signature data"))?;
+        Ok(EcdsaSig {
+            sig: secp256k1::ecdsa::Signature::from_der(signature)
+                .map_err(|_| encode::Error::ParseFailed("non-DER encoded signature"))?,
+            // NB: Since BIP-174 says "the signature as would be pushed to the stack from
+            // a scriptSig or witness" we should use a consensus deserialization and do
+            // not error on a non-standard values.
+            hash_ty: EcdsaSigHashType::from_u32_consensus(*sighash_byte as u32)
+        })
```

Comment:
This looks nearly identical to the implementation of `FromStr`. It'd be nicer to put the common part into a separate function and call it from both places. Maybe even `impl<'a> TryFrom<&[u8]> for EcdsaSig`

Reply:
There is no `TryFrom` in the current MSRV :(

Reply:
And they are not the same: one uses `from_u32_consensus` and the other `from_u32_standard`, which is an important difference. Also, errors are different, so moving this code to a shared function will result in having more code and not less.

Reply:
Oh, that's surprising difference. Why are they different?

Reply:
Since when we parsing string, we require user to provide us only correct values - and when parsing PSBT, we have to follow consensus serialization rules for signatures